src/libsyntax_pos/symbol.rs

```diff
@@ -1120,6 +1129,10 @@ impl !Sync for SymbolStr {}
 /// - `&*ss` is a `&str`;
 /// - `&ss as &str` is a `&str`, which means that `&ss` can be passed to a
 ///   function expecting a `&str`.
+///
+/// FIXME: This has no meaning anymore since the addition of `impl Add<char> for String`.
+/// Due to the outstanding Deref coercion issue this Deref implementation gets ignored.
+/// Issue: https://github.com/rust-lang/rust/issues/51916
```

Comment:
I assume the `Deref` impl is still used for plenty of other stuff. The only thing that changed is that it's not automatically used anymore when using `Add` or `AddAssign`. I would just remove this comment, I think.

Reply:
Done.